hles/transformation/src/main/scala/org/broadinstitute/monster/dap/dog/ResidentialEnvironmentTransformations.scala

```diff
@@ -58,38 +58,54 @@ object ResidentialEnvironmentTransformations {
           deLifetimeResidenceCount = Some(pastResidenceCount + currentResidenceCount.getOrElse(0L)),
           dePastResidenceZipCount = Some(pastZipCount),
           dePastResidenceCountryCount = Some(pastCountryCount),
-          dePastResidenceCountry1 = if (pastCountryCount > 1) {
-            parseCountry(rawRecord, "de_country_01", "de_country_01_dd")
-          } else {
-            parseCountry(rawRecord, "de_country_01_only", "de_country_01_only_dd")
-          },
-          dePastResidenceCountry2 =
-            if (pastCountryCount > 1) parseCountry(rawRecord, "de_country_02", "de_country_02_dd")
+          dePastResidenceCountry1 =
+            if (pastCountryCount > 1)
+              rawRecord
+                .getOptional("de_country_01_dd")
+                .orElse(rawRecord.getOptional("de_country_01_only_dd"))
             else None,
+          dePastResidenceCountry2 =
+            if (pastCountryCount > 2) rawRecord.getOptional("de_country_02_dd") else None,
```

Comment:
rather than copy-paste this line eighteen times, let's make a function for it and use that in all eighteen places instead.

Reply:
I renamed the function and moved the if (count check) and else None into the function.
I feel like this could be even further simplified if we just iterated over a range [1-10] to check for a certain prefix + suffix of RC variable for a given column in our table, something like:
`dePastResidenceCountry[N]        = getPastResidenceCountry(prefix="de_country_",suffix="_dd",range=2-10)
dePastResidenceCountry[N]Text = getPastResidenceCountry(prefix="de_country_",suffix=None,range=2-10)`
but I think what we have now may be better because its easier to read and understand the RC variable names we are checking. Let me know your thoughts

Reply:
i think abstracting over the prefix/suffix would be overkill, especially since we'd need to manually assign each one to its specific attribute anyway.